addons/account/models/account_move.py

```diff
@@ -3749,7 +3749,7 @@ def _generate_qr_code(self):
         """
         self.ensure_one()
 
-        if not self.display_qr_code:
+        if not self.display_qr_code or self.state == 'draft':
```

Comment:
Hmmm ... It's a pity we don't display the QR code when we could generate it safely, to have a more accurate preview. 

Let's rather not do this here and pass 
silent_errors=self.state == 'draft'
to the call to build_qr_code_base64() below. This way, if no error arises, we'll still show the QR code in draft. 

EDIT: it's not enough anyway. If I have a partner with an address that's not full (only city and country are set) and post the invoice before trying to get a preview, I still get an error in the preview. The right fix would probably to ensure we don't generate the QR in the preview if _check_for_qr_code_errors returns False.

Reply:
Wouldn't be better to let it as it is then ? So the user still knows what to fix in the preview before confirming the invoice.

And maybe, if we want to still have a preview in draft, add a context key for Switzerland 
https://github.com/odoo/odoo/blob/634c7bbe687c850b23e811b4b7e7b7da77313d99/addons/l10n_ch/models/res_bank.py#L381

in stable, doing something like 
```py
self.l10n_ch_qr_iban and not self._is_qr_reference(structured_communication) and not self.env.context.get('is_draft') 
```